src/net_processing.cpp

```diff
@@ -659,7 +662,8 @@ class PeerManagerImpl final : public PeerManager
     /** Potentially fetch blocks from this peer upon receipt of a new headers tip */
     void HeadersDirectFetchBlocks(CNode& pfrom, const Peer& peer, const CBlockIndex* pindexLast);
     /** Update peer state based on received headers message */
-    void UpdatePeerStateForReceivedHeaders(CNode& pfrom, const CBlockIndex *pindexLast, bool received_new_header, bool may_have_more_headers);
+    void UpdatePeerStateForReceivedHeaders(CNode& pfrom, Peer& peer, const CBlockIndex* pindexLast, bool received_new_header, bool may_have_more_headers)
```

Comment:
Do we really need to pass `Peer& peer` here? I know it's a pattern that's used in other functions too, but I don't understand why we wouldn't just call `GetPeerRef(pfrom.GetId())` instead. Unless I'm misunderstanding, `pfrom` and `peer` should always refer to the same node, so this seems unnecessarily confusing and bloating the function signature?

Reply:
We do this in a lot of places in `net_processing`, afaict we pass in a `Peer` when possible (because it is already available at the call site) otherwise we use `GetPeerRef`.